internal/controller/scheduler/scheduler.go

```diff
@@ -586,6 +586,18 @@ func (w *worker) Build(podSpec *corev1.PodSpec, skipCheckout bool, inputs buildI
 		w.logger.Debug("Applied podSpec patch from k8s plugin", zap.Any("patched", patched))
 	}
 
+	if skipCheckout {
```

Comment:
Given the complexity in this file I think there ought to be a comment added that explains what's going on:

```suggestion
	// PodSpecPatch can be used to add a container named "checkout" even when
	// checkout has been disabled, so this removes it.
	if skipCheckout {
```

Reply:
Thanks for the feedback. Changes implemented!